pkg/expression/builtin_time_test.go

```diff
@@ -2592,6 +2592,40 @@ func TestTimestampAdd(t *testing.T) {
 		{"MONTH", 10, "2024-10-31", "2025-08-31 00:00:00"},
 		{"MONTH", 1, "2024-11-30", "2024-12-30 00:00:00"},
 		{"MONTH", 13, "2024-11-30", "2025-12-30 00:00:00"},
+
+		// issue 54908
+		{"MONTH", 0, "2024-09-01", "2024-09-01 00:00:00"},
+		{"MONTH", -10, "2024-09-01", "2023-11-01 00:00:00"},
+		{"MONTH", -2, "2024-04-28", "2024-02-28 00:00:00"},
+		{"MONTH", -2, "2024-04-29", "2024-02-29 00:00:00"},
+		{"MONTH", -2, "2024-04-30", "2024-02-29 00:00:00"},
+		{"MONTH", -1, "2024-03-28", "2024-02-28 00:00:00"},
+		{"MONTH", -1, "2024-03-29", "2024-02-29 00:00:00"},
+		{"MONTH", -1, "2024-03-30", "2024-02-29 00:00:00"},
+		{"MONTH", -1, "2024-03-31", "2024-02-29 00:00:00"},
+		{"MONTH", -1, "2024-03-25", "2024-02-25 00:00:00"},
+		{"MONTH", -12, "2024-03-31", "2023-03-31 00:00:00"},
+		{"MONTH", -13, "2024-03-31", "2023-02-28 00:00:00"},
+		{"MONTH", -14, "2024-03-31", "2023-01-31 00:00:00"},
+		{"MONTH", -24, "2024-03-31", "2022-03-31 00:00:00"},
+		{"MONTH", -25, "2024-03-31", "2022-02-28 00:00:00"},
+		{"MONTH", -26, "2024-03-31", "2022-01-31 00:00:00"},
+		{"MONTH", -1, "2024-02-25", "2024-01-25 00:00:00"},
+		{"MONTH", -11, "2025-02-28", "2024-03-28 00:00:00"},
+		{"MONTH", -12, "2025-02-28", "2024-02-28 00:00:00"},
+		{"MONTH", -13, "2025-02-28", "2024-01-28 00:00:00"},
+		{"MONTH", -11, "2024-02-29", "2023-03-29 00:00:00"},
+		{"MONTH", -12, "2024-02-29", "2023-02-28 00:00:00"},
+		{"MONTH", -13, "2024-02-29", "2023-01-29 00:00:00"},
+		{"MONTH", -11, "2023-02-28", "2022-03-28 00:00:00"},
+		{"MONTH", -12, "2023-02-28", "2022-02-28 00:00:00"},
+		{"MONTH", -13, "2023-02-28", "2022-01-28 00:00:00"},
+		{"MONTH", -2, "2023-02-28", "2022-12-28 00:00:00"},
+		{"MONTH", -14, "2023-02-28", "2021-12-28 00:00:00"},
+		{"MONTH", -3, "2023-03-20", "2022-12-20 00:00:00"},
+		{"MONTH", -3, "2023-03-31", "2022-12-31 00:00:00"},
+		{"MONTH", -15, "2023-03-20", "2021-12-20 00:00:00"},
+		{"MONTH", -15, "2023-03-31", "2021-12-31 00:00:00"},
```

Comment:
Add tests to test from `2020-02-29` to `2021-02-28` and from `2020-02-29` to `2019-02-28`

Reply:
And add some invalid input test cases

Reply:
> And add some invalid input test cases

done